pkg/capture/capture_manager.go

```diff
@@ -192,17 +220,30 @@ func (cm *Manager) Status(ctx context.Context, ifaces ...string) (statusmap map[
 }
 
 // Update the configuration for all (or a set of) interfaces
-func (cm *Manager) Update(ctx context.Context, ifaces config.Ifaces) {
+func (cm *Manager) Update(ctx context.Context, ifaces config.Ifaces) (enabled, updated, disabled []string, err error) {
+	// Validate the config before doing anything else
+	err = ifaces.Validate()
+	if err != nil {
+		return
+	}
 
 	// Build set of interfaces to enable / disable
 	var (
-		ifaceSet                    = make(map[string]struct{})
-		enableIfaces, disableIfaces []string
+		ifaceSet                                  = make(map[string]struct{})
+		enableIfaces, updateIfaces, disableIfaces []string
 	)
-	for iface := range ifaces {
+	for iface, cfg := range ifaces {
 		ifaceSet[iface] = struct{}{}
 		if _, exists := cm.captures.Get(iface); !exists {
 			enableIfaces = append(enableIfaces, iface)
+		} else {
+			updatedCfg := cfg
+			runtimeCfg := cm.lastAppliedConfig[iface]
+
+			// take care of parameter updates to an interface that exists already
+			if !reflect.DeepEqual(updatedCfg, runtimeCfg) {
```

Comment:
No strong feelings, but we need an equality check. Background: in most cases, we want to update goProbe's runtime config from the config file that it parsed during start up, but which has changed while it was running.

If we forced a reload for every interface in the provided update configuration, we'd be unnecessarily writing out interfaces that don't need a reload because their parameters didn't change.